extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/Job.java

```diff
@@ -111,8 +112,16 @@ public static void scheduleRange(String thingUID, AstroThingHandler astroHandler
         if (thingNull || astroHandlerNull || rangeNull || channelIdNull) {
             return;
         }
-        scheduleEvent(thingUID, astroHandler, range.getStart(), EVENT_START, channelId);
-        scheduleEvent(thingUID, astroHandler, range.getEnd(), EVENT_END, channelId);
+        
+        //add 1 second to the last scheduled event if both the events comprise same time instant
+        int secondsPadding = 1;
```

Comment:
Maybe I missed something, but wouldn't it be much shorter when using DateUtils directly to compare those two dates?

```java
Calendar end = range.getEnd();
if (DateUtils.truncatedEquals(range.getStart(), end, SECOND) {
    end.add(SECOND, 1);
}
```

regarding the padding: either inline the value directly or declare a constant field for it.